app/services/users/archive_by_email.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Users
+  class ArchiveByEmail
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :user
+
+    def call
+      ApplicationRecord.transaction do
+        users_with_matching_email.each do |user_with_matching_email|
+          move_applications(from_user: user_with_matching_email, to_user: user)
+          if Feature.ecf_api_disabled?
+            move_participant_id_changes(from_user: user_with_matching_email, to_user: user)
+            user.participant_id_changes.find_or_create_by!(from_participant_id: user_with_matching_email.ecf_id, to_participant_id: user.ecf_id)
+          end
+        end
+      end
+      users_with_matching_email.each do |user_with_matching_email|
```

Comment:
Is there any reason why this block is outside the transaction? As you're doing a loop in `users_with_matching_email` twice..

Reply:
the archiver has it's own transaction